arangod/VocBase/LogicalView.cpp

```diff
@@ -121,6 +129,50 @@ LogicalView::LogicalView(
 
   auto view = viewFactory(vocbase, definition, isNew, planVersion, preCommit);
 
+  ////////////////////////////////////////////////////////////////////////////
+  // AT THIS POINT the view is already created but does not have links
+  // from the properties set-
+  //
+  // Add them now manually for arangosearch, because they unable to do it.
+  ////////////////////////////////////////////////////////////////////////////
+
+  if (view && viewType.compare("arangosearch") == 0) {
```

Comment:
view-implementation specific code should reside inside the view implementation and not inside generic code